### ISSUE TITLE
Trusted Book Providers

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#5792]
- [x] #5271 Add basic flow for book providers
- [ ] Project Gutenberg
    - Titles in catalogue: [~65,000](https://en.wikipedia.org/wiki/Project_Gutenberg)
    - Editions in Open Library: [873](http://openlibrary.org/query.json?type=/type/edition&publishers=Project%20Gutenberg&limit=5000)
    - Records in IA: [59,105](https://archive.org/details/gutenberg)
    - Works in Open Library: [535](https://openlibrary.org/search?q=publisher%3A%22Project+Gutenberg%22&mode=everything)
    - [x] Display read button when provider ID present
    - [x] Custom download options
    - [ ] Add Gutenberg ID to all the `publisher:Project Gutenberg` books currently in Open Library
    - [ ] Import missing records
    - [ ] Set up automatic import pipeline
- [ ] LibriVox
    - Titles in catalogue: [~15,000](https://librivox.org/2021/02/14/librivox-celebrates-15000-audiobooks/)
    - Editions in Open Library: [598](http://openlibrary.org/query.json?type=/type/edition&publishers=LibriVox&limit=5000)
    - Works in Open Library: [423](https://openlibrary.org/search?q=publisher%3ALibriVox&mode=everything)
    - Records in IA: [16,057](https://archive.org/details/librivoxaudio)
    - [x] Display read button when provider ID present
    - [x] Custom download options
    - [x] Add proper LibriVox numeric ID to all the `publisher:LibriVox` books currently in Open Library
    - [ ] #8105
    - [ ] Get/crawl a dump of all librivox IDs
    - [ ] Bulk import all librivox IDs
    - [ ] Set up automatic import pipeline
- [x] Standard Ebooks
    - Titles in catalogue: [589](https://standardebooks.org/ebooks?page=50)
    - Editions in Open Library: [10](http://openlibrary.org/query.json?type=/type/edition&publishers=Standard%20Ebooks&limit=5000)
    - Works in Open Library: [11](https://openlibrary.org/search?q=publisher%3A%22Standard+Ebooks%22&mode=everything)
    - [x] Display read button when provider ID present
    - [x] Custom download options
    - [x] #3982 
    - [x] Import all standard ebooks books
        - [x] Blocked by #5993
        - [x] #5994
        - [x]  #6070
- [ ] OpenStax
    - Titles in catalogue: [42](https://openstax.org/about)
    - Works in Open Library: [~50](https://openlibrary.org/search?q=openstax&mode=everything)
    - [ ] Tidy up the records
         - [x] #6152
         - [x] Remove accidental `openstax_id` field
    - [x] Add proper openstax ID to every record
    - [x] #6071 Display read button when provider ID present
    - [x] #6071 Custom download options
    - [x] #5577 Import missing records
    - [ ] Set up automatic import pipeline
- [ ] Wikisource
    - Titles in catalogue: [~500,000](https://en.wikisource.org/wiki/Main_Page) in English alone!
    - No real presence in OL yet
    - [x] Create identifier field for Wikisource
        - Big Question: wikisource IDs are language specific (eg note the `en` in https://en.wikisource.org/wiki/Goody_Two-Shoes_(1888) ) :grimacing: how should we deal with that?
        - Big Answer: Apparently URLs like this work! https://wikisource.org/wiki/en:The_Annotated_Strange_Case_of_Dr_Jekyll_and_Mr_Hyde
    - [ ] Display read button when provider ID present
    - [ ] Custom download options
    - [ ] Import missing records
    - [ ] Set up automatic import pipeline

Usability Improvements:
- [x] Add option to open a work to a specific provided edition (eg https://openlibrary.org/works/OL413631W/Roughing_It_in_two_volumes._2_2?edition=gutenberg%3A8583 )
- [x] Add option to prefer a provider in search results (eg https://openlibrary.org/search?mode=everything&q=publisher%3Aproject+gutenberg&sort=random&providerPref=gutenberg )
- [x] #5991
- [x] #5992
- [x] #5803 Not working on lists! Replace old availability.js with server-side LoanStatus
- [ ] #8548
- [ ] #8546
- [ ] #8547

Marketing/discovery:
- [x] #5837
- [ ] Add collection page (for each provider?)

Bugs:
- [x] #5793 
- [x]  #5806
- [x] #5802
- [ ] OpenStax slugs redirect to latest edition for older editions :( Need to make metadata request. See https://github.com/internetarchive/openlibrary/pull/6138#pullrequestreview-883750756
- [x] #6379

**Comment by mekarpeles** [comment]
Other sources:

https://onlinebooks.library.upenn.edu/
https://open.umn.edu/opentextbooks
https://www.intechopen.com/popular-books
https://cain.math.gatech.edu/textbooks/onlinebooks.html
https://openlibra.com/en/
https://www.oapen.org/
https://freecomputerbooks.com/
https://unglue.it/lists/popular
http://users.erols.com/jonwill/freebooks.htm

**Comment by cdrini** [comment]
Splitting this out into separate issue for each provider:

- #8543
- #8544
- #8545

Otherwise this issue is closed!
